mmv1/api/resource_test.go

```diff
@@ -198,3 +198,105 @@ func TestResourceServiceVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestLeafProperties(t *testing.T) {
```

Comment:
would it be useful to have tests for non-nested leaf properties to cover the base case?

Reply:
Yea that's fair, added a base case